.github/workflows/build+release.yml

```diff
@@ -0,0 +1,66 @@
+name: Build and upload release
+
+on:
+    workflow_dispatch: # trigger manually
+        inputs:
+            tag_override:
+                description: "Tag"
+                type: string
+                required: false
+            release_notes:
+                description: "Release notes"
+                type: string
+                required: true
+
+env:
+    node_version: "16.x"
+
+jobs:
+    build-and-upload-release:
+        runs-on: ubuntu-latest
+        steps:
+            - name: Apply tag override and parse ref
+              uses: bisgardo/github-action-parse-ref@v1
+              id: ref
+              with:
+                  ref: "${{github.event.inputs.tag_override || github.ref}}"
+                  default-ref-type: tags
+            - name: Fail if ref is not a tag
+              if: "steps.ref.outputs.ref-type != 'tags'"
+              run: exit 1
+            - name: Parse tag name
+              uses: bisgardo/github-action-regex-parse@v1
+              id: tag
+              with:
+                  pattern: "(?<name>(?<package>.*?)/(?<version>.*))"
+                  input: "${{steps.ref.outputs.ref-name}}"
+            - name: Checkout subproject
+              # Clone repository into temp directory and copy the subproject into the workspace
+              # (cannot use 'actions/checkout' because that doesn't allow checkout out the repo outside the workspace dir;
+              # see 'https://github.com/actions/checkout/issues/327' - alternatively we could use a temp dir inside the workspace).
+              run: |
+                  git clone --branch="${{steps.tag.outputs.name}}" "${{github.server_url}}/${{github.repository}}" "${{runner.temp}}/repo"
+                  cp -rT "${{runner.temp}}/repo/packages/${{steps.tag.outputs.package}}" .
```

Comment:
I don't understand why you need to do this? Can't you just checkout and then set the working-directory for the subsequent steps? :thinking:

Reply:
Unfortunately no:
1. You cannot change workdir for the remaining steps, it has to be set for all of them. And it can only be done for `run` steps, not actions (i.e. `uses`).
2. If you have the entire repo structure intact, yarn will expect local libraries that it depends on to have been built locally and not fetch them from npm.